prow/cmd/deck/main.go

```diff
@@ -1172,29 +1202,33 @@ func handleRerun(prowJobClient prowv1.ProwJobInterface, createProwJob bool) http
 		// On Prow instances that require auth even for viewing Deck this is okayish, because the Prowjob UUID
 		// is hard to guess
 		// Ref: https://github.com/kubernetes/test-infra/pull/12827#issuecomment-502850414
-		if createProwJob {
-			if r.Method != http.MethodPost {
-				http.Error(w, "request must be of type POST", http.StatusMethodNotAllowed)
+		if r.Method == http.MethodPost {
+			if !createProwJob {
+				http.Error(w, "Direct rerun feature is not yet enabled", http.StatusMethodNotAllowed)
+				return
+			}
+			login, err := r.Cookie("github_login")
+			if err == http.ErrNoCookie {
+				http.Redirect(w, r, "/github-login?dest=%2F?rerun=work_in_progress", http.StatusFound)
+				return
+			} else if err != nil {
+				http.Error(w, fmt.Sprintf("Error retrieving GitHub cookie: %v", err), http.StatusInternalServerError)
+				logrus.WithError(err).Errorf("Error retrieving GitHub cookie")
+				return
+			}
+			if !canTriggerJob(login.Value, cfg().Deck.RerunAuthConfig) {
```

Comment:
Instead of passing in cfg to this function you can pass in a 

```golang

type func() config.RerunAuthConfig authCfg // use this in handleRerun

cfgGetter := func() config.RerunAuthConfig { return cfg().Deck.RerunAuthConfig }
gziphandler.GzipHandler(handleRerun(prowJobClient, o.rerunCreatesJob, cfgGetter))
```

That way your unit tests can just define this struct, not the whole config and this will ensure that we don't sneak in unexpected usage of the config